src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueue.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Enable.Extensions.Queuing.AzureServiceBus.Internal
+{
+    internal class AzureServiceBusQueue : IDisposable
+    {
+        private int _referenceCount = 1;
+        private bool _disposed = false;
+
+        public AzureServiceBusQueue(string connectionString, string queueName)
+        {
+            Receiver = new MessageReceiver(
+                connectionString,
+                queueName,
+                ReceiveMode.PeekLock);
+
+            Sender = new MessageSender(connectionString, queueName);
+        }
+
+        public IMessageSender Sender { get; }
+        public IMessageReceiver Receiver { get; }
+
+        public int AddReference()
```

Comment:
This method name is a little unclear. Can we call this `IncrementReferenceCount` instead? This makes it clearer what this method does and makes it consistent with the equivalent in-memory queue client.

Reply:
Done.